R/zzz.R

```diff
@@ -8,6 +8,21 @@
   vctrs::s3_register("butcher::axe_fitted", "workflow")
 
   vctrs::s3_register("generics::required_pkgs", "workflow")
+
+  # Can't use `rlang::is_installed()` at all, as that doesn't work when
+  # called from `.onLoad()` for some reason. Instead, rely on `packageVersion()`
+  # erroring when the package isn't installed.
+  has_at_least_version <- function(pkg, version) {
+    tryCatch(
+      expr = utils::packageVersion(pkg) >= version,
+      error = function(cnd) FALSE
+    )
+  }
```

Comment:
I think we are going to be in the same "bad" scenario Julia found herself in with using workflows in vetiver, where she doesn't need tune to be installed but she wanted the `required_pkgs()` method for workflows to be registered.

This might work a little better:

```suggestion
  # - If tune isn't installed, register the method (`packageVersion()` will error here)
  # - If tune >= 0.1.6.9001 is installed, register the method
  should_register_tune_args_method <- tryCatch(
    expr = utils::packageVersion("tune") >= "0.1.6.9001",
    error = function(cnd) TRUE
  )
```

Then you can do

```r
if (should_register_tune_args_method) {

}
```